src/libs/runtime-utils.js

```diff
@@ -177,6 +179,19 @@ export const appIsSettingUp = app => {
   return app && (app.status === 'PROVISIONING' || app.status === 'PRECREATING')
 }
 
+export const currentPersistentDiskIncludingUnattached = (apps, galaxyDataDisks) => {
+  const currentGalaxyApp = currentAppIncludingDeleting(apps)
+  const currentDataDiskName = currentGalaxyApp?.diskName
+  const attachedDataDiskNames = _.without([undefined], _.map(app => app.diskName, apps))
+  const currentDatDisk = currentDataDiskName ?
+    _.find({ name: currentDataDiskName }, galaxyDataDisks) :
+    _.last(_.sortBy('auditInfo.createdDate', _.filter(({ name, status }) => status !== 'Deleting' && !_.includes(name, attachedDataDiskNames), galaxyDataDisks)))
```

Comment:
To make sure I understand, would the line above get executed if a user deleted their Galaxy app but kept the disks?

Reply:
yes, that's correct. it would only get executed in that case. the other case is that the user has a current app that's either active or in deleting status so the disk is either attached or detaching

Reply:
i'll add a comment to make this a bit more clear

Reply:
The comments are helpful. Thank you.